Not compiling with Apple LLVM version 4.2

Fails to compile citing an error:

```
[ 41%] Building CXX object gnuradio-core/src/lib/CMakeFiles/gnuradio-core.dir/filter/gr_fir_ccf_x86.cc.o
Building C object gnuradio-core/src/lib/CMakeFiles/gnuradio-core.dir/filter/sse_debug.c.o
brew: superenv removed: -I/usr/local/include
brew: superenv removed: -I/usr/local/include
brew: superenv removed: -I/usr/local/include
[ 41%] Building C object gnuradio-core/src/lib/CMakeFiles/gnuradio-core.dir/filter/float_dotprod_sse64.S.o
brew: superenv removed: -I/usr/local/include
/tmp/gnuradio-KXU7/gnuradio-3.6.1/gnuradio-core/src/lib/filter/float_dotprod_sse64.S:57:2: error: unknown directive
 .version "01.01"
 ^
make[2]: *** [gnuradio-core/src/lib/CMakeFiles/gnuradio-core.dir/filter/float_dotprod_sse64.S.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [gnuradio-core/src/lib/CMakeFiles/gnuradio-core.dir/all] Error 2
make: *** [all] Error 2
==> Configuration
HOMEBREW_VERSION: 0.9.4
HEAD: bd309c49d29e2151ec86cfb04919783834b7ca74
CPU: 8-core 64-bit ivybridge
OS X: 10.8.3-x86_64
Xcode: 4.6.2
CLT: 4.6.0.0.1.1358221012
X11: 2.7.4 => /opt/X11
==> ENV
CC: cc
CXX: c++
MAKEFLAGS: -j8
CMAKE_PREFIX_PATH: /usr/local/opt/readline:/usr/local/opt/sqlite:/usr/local/opt/gettext:/usr/local/opt/libffi:/usr/local/opt/icu4c:/usr/local
CMAKE_INCLUDE_PATH: /opt/X11/include/freetype2:/usr/include/libxml2:/opt/X11/include
CMAKE_LIBRARY_PATH: /opt/X11/lib
PKG_CONFIG_PATH: /usr/local/opt/sqlite/lib/pkgconfig:/usr/local/opt/cppunit/lib/pkgconfig:/usr/local/opt/gsl/lib/pkgconfig:/usr/local/opt/fftw/lib/pkgconfig:/usr/local/opt/pcre/lib/pkgconfig:/usr/local/opt/libffi/lib/pkgconfig:/usr/local/opt/glib/lib/pkgconfig:/usr/local/opt/libtiff/lib/pkgconfig:/usr/local/opt/libpng/lib/pkgconfig:/usr/local/opt/gdk-pixbuf/lib/pkgconfig:/usr/local/opt/icu4c/lib/pkgconfig:/usr/local/opt/freetype/lib/pkgconfig:/usr/local/opt/fontconfig/lib/pkgconfig:/usr/local/opt/harfbuzz/lib/pkgconfig:/usr/local/opt/pixman/lib/pkgconfig:/usr/local/opt/cairo/lib/pkgconfig:/usr/local/opt/pango/lib/pkgconfig:/usr/local/opt/atk/lib/pkgconfig:/usr/local/opt/gtk+/lib/pkgconfig:/usr/local/opt/pygobject/lib/pkgconfig:/usr/local/opt/py2cairo/lib/pkgconfig:/usr/local/opt/pygtk/lib/pkgconfig:/usr/local/opt/sdl/lib/pkgconfig:/usr/local/opt/libusb/lib/pkgconfig:/usr/local/opt/orc/lib/pkgconfig:/usr/local/opt/qt/lib/pkgconfig
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Library/ENV/pkgconfig/10.8:/opt/X11/lib/pkgconfig:/opt/X11/share/pkgconfig
ACLOCAL_PATH: /usr/local/opt/gettext/share/aclocal:/usr/local/share/aclocal:/opt/X11/share/aclocal
OBJC: cc
PATH: /usr/local/Library/ENV/4.3:/usr/local/opt/cmake/bin:/usr/local/opt/sqlite/bin:/usr/local/opt/gdbm/bin:/usr/local/opt/python/bin:/usr/local/opt/cppunit/bin:/usr/local/opt/gsl/bin:/usr/local/opt/fftw/bin:/usr/local/opt/pcre/bin:/usr/local/opt/swig/bin:/usr/local/opt/gettext/bin:/usr/local/opt/glib/bin:/usr/local/opt/jpeg/bin:/usr/local/opt/libtiff/bin:/usr/local/opt/libpng/bin:/usr/local/opt/gdk-pixbuf/bin:/usr/local/opt/icu4c/bin:/usr/local/opt/freetype/bin:/usr/local/opt/fontconfig/bin:/usr/local/opt/harfbuzz/bin:/usr/local/opt/cairo/bin:/usr/local/opt/pango/bin:/usr/local/opt/gtk+/bin:/usr/local/opt/pygobject/bin:/usr/local/opt/pygtk/bin:/usr/local/opt/sdl/bin:/usr/local/opt/orc/bin:/usr/local/opt/sip/bin:/usr/local/opt/qt/bin:/usr/local/opt/pyqt/bin:/opt/X11/bin:/usr/bin:/bin:/usr/sbin:/sbin

Error: gnuradio did not build
```

It seems that this is an issue where gnuradio fails to compile with clang: http://trac.macports.org/ticket/36453

I'm not sure how this worked before but perhaps it was before clang became the default compiler. I'm going to fix this and initiate a pull request.

--- a/gnuradio.rb
+++ b/gnuradio.rb
@@ -43,6 +43,12 @@ class Gnuradio < Formula
   end
 
   def install
+
+	  # Force compilation with gcc-4.2
+	  ENV['CC'] = '/usr/local/bin/gcc-4.2'
+	  ENV['LD'] = '/usr/local/bin/gcc-4.2'
+	  ENV['CXX'] = '/usr/local/bin/g++-4.2'
+
     mkdir 'build' do
       args = ["-DCMAKE_PREFIX_PATH=#{prefix}", "-DQWT_INCLUDE_DIRS=#{HOMEBREW_PREFIX}/lib/qwt.framework/Headers"] + std_cmake_args
       args << '-DENABLE_GR_QTGUI=OFF' unless ARGV.include?('--with-qt')

--- a/gnuradio.rb
+++ b/gnuradio.rb
@@ -6,7 +6,7 @@ class Gnuradio < Formula
   sha1 'ee4c40b366d94887511eefa6ce52df4bdbc9d105'
   head 'git://gnuradio.org/gnuradio.git'
 
-  depends_on 'apple-gcc4.2' => :build
+  depends_on 'apple-gcc42' => :build
   depends_on 'cmake' => :build
   depends_on 'Cheetah' => :python
   depends_on 'lxml' => :python

--- a/gnuradio.rb
+++ b/gnuradio.rb
@@ -6,6 +6,7 @@ class Gnuradio < Formula
   sha1 'ee4c40b366d94887511eefa6ce52df4bdbc9d105'
   head 'git://gnuradio.org/gnuradio.git'
 
+  depends_on 'apple-gcc4.2' => :build
   depends_on 'cmake' => :build
   depends_on 'Cheetah' => :python
   depends_on 'lxml' => :python

--- a/gr-baz.rb
+++ b/gr-baz.rb
@@ -12,8 +12,9 @@ class GrBaz < Formula
 
   def install
     args = ["--prefix=#{prefix}"]
-    system "autoreconf -i"
-    system "./configure", *args
+    system "sh bootstrap"
+    system "sh configure", *args
+    system "make"
     system "make install"
   end
 end

--- a/gr-baz.rb
+++ b/gr-baz.rb
@@ -10,11 +10,41 @@ class GrBaz < Formula
   depends_on 'libusb'
   depends_on 'gnuradio'
 
+  def patches
+    DATA
+  end
+
   def install
     args = ["--prefix=#{prefix}"]
     system "sh bootstrap"
-    system "sh configure", *args
+    system "./configure", *args
     system "make"
     system "make install"
   end
 end
+
+__END__
+diff --git a/bootstrap b/bootstrap
+index 2412a7a..8f5799f 100644
+--- a/bootstrap
++++ b/bootstrap
+@@ -25,5 +25,5 @@ rm -fr config.cache autom4te*.cache
+ aclocal -I config
+ autoconf
+ autoheader
+-libtoolize --automake -c -f
++glibtoolize --automake -c -f
+ automake --add-missing -c -f -Wno-portability
+diff --git a/config/gr_standalone.m4 b/config/gr_standalone.m4
+index 35cb789..0ad91ce 100644
+--- a/config/gr_standalone.m4
++++ b/config/gr_standalone.m4
+@@ -29,7 +29,7 @@ dnl get called too late to be useful.
+ m4_define([GR_STANDALONE],
+ [
+   AC_CONFIG_SRCDIR([config/gr_standalone.m4])
+-  AM_CONFIG_HEADER(config.h)
++  AC_CONFIG_HEADER(config.h)
+ 
+   dnl Remember if the user explicity set CXXFLAGS
+   if test -n "${CXXFLAGS}"; then

